promisasu/Portal
# Validate Configuration

**Github Issue**
#307

**Taiga Issue**
557

**What did you change?**
* Replaced Joi with `JSON Lint` and a json schema. 
* Added validation before `npm start`
* Renamed `npm run check-dependencies` to `npm run check`

**How can we test?**
run `npm install`
make one of the changes list below
run `npm start`

**1.** rename/rename `config.json`

Linter should print the message
```
Error: ENOENT: no such file or directory, open 'config.json'
```

**2.** Make `config.json` invalid (remove a comma, add an extra comma, remove or add quotes)

Linter should print a message like
```
[Error: Parse error on line 5:
...    "port": 3000,  },  "api": {    "h
----------------------^
Expecting 'STRING', got '}']
```
It should list the line/column that has the issue

**3.** Remove a required attribute in `config.json`

Linter should print a message like
```
Validation Errors:

Property is required
uri: urn:uuid:2e7d517e-f598-49b3-9cf3-9c642ba016e7#/database/salt
schemaUri: urn:uuid:970c467f-ea44-48d7-a97e-99a4b73c7778#/properties/database/properties/salt
attribute: required
details: true
```
